iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -132,6 +133,16 @@ private async Task InitializeAndSetupClientAsync(CancellationToken cancellationT
                     cancellationToken: cancellationToken);
                 _logger.LogDebug($"The client instance has been opened.");
 
+                // The client retrieves twin values when it is initialized.
+                await RetryOperationHelper.RetryTransientExceptionsAsync(
+                    operationName: "GetTwin",
+                    asyncOperation: async () => await RetrieveTwinAsync(cancellationToken),

```

Comment:
You can update `RetryTransientExceptionsAsync` to accept `Func<Task<T>> asyncOperation` and then you can call `s_deviceClient.GetTwinAsync()` in here directly.
RetryTransientExceptionsAsync checks if the client is connected before performing the opration, so you can avoid having to add that check twice.

Reply:
> You can update `RetryTransientExceptionsAsync` to accept `Func<Task<T>> asyncOperation` and then you can call `s_deviceClient.GetTwinAsync()` in here directly. RetryTransientExceptionsAsync checks if the client is connected before performing the opration, so you can avoid having to add that check twice.

modified